### PR TITLE
Feat/keymap legend style

### PR DESCRIPTION
**Pull request by seaerchin**
## Problem 
This PR addresses #1183, which is that  users want to be able to have different keymap legends. however, there is no such setting available now. 

## Solution
This PR addresses this through 
1. adding settings 
2. adding config
3. updating the DOM upon selection 

## Screenshots 
**Settings menu view**
image

**Blank**
image

**Uppercase**
image

### CODE REVIEW

**Comment by Miodec** [comment]
I think the buttons in the settings page should be: `lowercase`, `uppercase`, `blank`.

**Comment by seaerchin** [comment]
> I think the buttons in the settings page should be: `lowercase`, `uppercase`, `blank`.

this is as opposed to `lowercase`, `UPPERCASE` and ` ` correct? i thought it'd be a good thematic fit to have it that way but i'll change if it's an issue

**Comment by Miodec** [comment]
Yeah thats what i mean. All other settings page buttons are lowercase too.

File: static/keymap/legend/_list.json [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{
+    "lowercase": "lowercase",
+    "uppercase": "uppercase",
+    "blank":  "blank"
+}
```
Comment (Miodec) [inline static/keymap/legend/_list.json:5]: Also i don't fully understand the separate file approach here. Seems a bit overcomplicated compared to other configs.

File: src/js/config.js [hunk 2]
```diff
@@ -1264,22 +1266,41 @@ export function setKeymapMode(mode, nosave) {
...
+  }
+
+  // Update and save to cookie for persistence
+  $(".keymapLegendStyle").addClass(style);
+  config.keymapLegendStyle = style;
+  if (!nosave) saveToCookie();
+}
+
 export function setKeymapStyle(style, nosave) {
+  console.log("keymap style", style);
```
Comment (Miodec) [inline src/js/config.js:1297]: Please remove console log.

File: static/keymap/legend/_list.json (hunk 1 above)
Comment (seaerchin) [reply static/keymap/legend/_list.json:5]: i wanted a single source of truth throughout the application for the keymap legends settings but if you feel that it's unneeded/too complicated, i'll remove it!

Comment (Miodec) [reply static/keymap/legend/_list.json:5]: We don't do this for other settings so I think I would prefer it to be continuous across all settings.

Comment (seaerchin) [reply static/keymap/legend/_list.json:5]: sure, i'll change this

**Pull request merged**